### ISSUE TITLE
AY19/20 Sem 2 Q2

### ISSUE THREAD

**Issue by e0588269** [nus-cs2030/2021-s2#308]
hello! im quite confused about how heap and stack actually work, so I tried Q2 of AY19/20. Could someone help to correct me please, thank you :)

[q2.pdf](https://github.com/nus-cs2030/2021-s2/files/6363483/q2.pdf)
![image](https://user-images.githubusercontent.com/80536958/115834203-8838f700-a447-11eb-81dd-89f742c851b0.png)

**Comment by gableejh** [comment]
the first one looks good, as for the second and third one, I did it this way

![image](https://user-images.githubusercontent.com/77229491/115840796-a3f3cb80-a44e-11eb-84ab-38eab3f6e73d.png)

for your `Library lib = new Lib().loanBook(bk, br);`, I don't think it can be assigned this way since loanBook is a void method that doesn't return anything

**Comment by yuyitan** [comment]
In the question, it is written "the variables shown in the stack frame of the main method are declared accordingly from top to bottom" thus you can just follow the same order when you are initialising your classes.

**Comment by deniselam10** [comment]
For the second one, is it possible to put 'Depot d = new Depot();' before the addressee instead?

**Comment by Englishificational** [comment]
> For the second one, is it possible to put 'Depot d = new Depot();' before the addressee instead?

I don't think so. The stack order would get messed up since you need to declare in the order apcd.

**Comment by deniselam10** [comment]
> > For the second one, is it possible to put 'Depot d = new Depot();' before the addressee instead?
> 
> I don't think so. The stack order would get messed up since you need to declare in the order apcd.

I see, thanks for your help!
